### PR TITLE
Calendar

### PR DESCRIPTION
**Pull request by ghost**
- [DEMO LINK](https://WaveOfDandelions.github.io/layout_calendar/)

### CODE REVIEW

File: src/styles/main.scss [hunk 1]
```diff
@@ -1,3 +1,73 @@
...
   margin: 0;
 }
+
+$grey: #eee;
+$square-side: 100px;
+$calendar-gap: 1px;
+$calendar-day-border: 1px;
+$calendar-day-space: $square-side + $calendar-gap;
+$calendar-padding: 10px;
+$calendar-width: ($square-side + $calendar-gap) * 7;
```
Comment (mvjl000) [inline src/styles/main.scss]: Create a separate file for variables ;)

File: src/styles/main.scss [hunk 2]
```diff
@@ -1,3 +1,73 @@
...
+
+.calendar {
+  width: $calendar-width;
+  margin: 0 auto;
```
Comment (mvjl000) [inline src/styles/main.scss]: The calendar should be centered both horizontally and vertically. Maybe try to add display flex to parent, which is body

File: src/styles/main.scss [hunk 3]
```diff
@@ -1,3 +1,73 @@
...
+  display: flex;
+  flex-wrap: wrap;
+  gap: $calendar-gap;
+  padding: $calendar-padding;
+  padding-top: $calendar-padding * 3;
```
Comment (mvjl000) [inline src/styles/main.scss]: I don't think this two variables make sense

File: src/styles/main.scss [hunk 4]
```diff
@@ -1,3 +1,73 @@
...
+    justify-content: center;
+    align-items: center;
+    transition: 0.5s;
+  }
+
+  @for $index from 1 through 31 {
+    &__day:nth-child(#{$index})::before {
+      content: "#{$index}";
+      font-family: Arial, sans-serif;
+      font-size: 30px;
```
Comment (mvjl000) [inline src/styles/main.scss]: This should be inside of day. Inside of loops, put only the properties that are relying on loop - in this case content.

File: src/styles/main.scss [hunk 5]
```diff
@@ -1,3 +1,73 @@
...
+    }
+  }
+
+  @for $index from 1 through 31 {
+    &__day:nth-child(#{$index}):hover {
+      background-color: pink;
+      cursor: pointer;
```
Comment (mvjl000) [inline src/styles/main.scss]: I would like you to move `cursor: pointer` up, so it's located inside of __day

File: src/styles/main.scss [hunk 6]
```diff
@@ -1,3 +1,73 @@
...
+  &__day {
+    box-sizing: border-box;
+    background-color: $grey;
+    width: $square-side;
+    height: $square-side;
+    border: $calendar-day-border solid black;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    transition: 0.5s;
```
Comment (mvjl000) [inline src/styles/main.scss]: Make sure to list all styles that you apply transition to - don't use `all` either.

**Review by mvjl000 — CHANGES_REQUESTED** [review]

File: src/styles/main.scss (hunk 1 above)
Comment (WaveOfDandelions) [reply src/styles/main.scss]: > Create a separate file for variables ;)

Done

File: src/styles/main.scss (hunk 2 above)
Comment (WaveOfDandelions) [reply src/styles/main.scss]: > The calendar should be centered both horizontally and vertically. Maybe try to add display flex to parent, which is body

Done, I was set 100vh for body, and use flex alignment

File: src/styles/main.scss (hunk 5 above)
Comment (WaveOfDandelions) [reply src/styles/main.scss]: > I would like you to move `cursor: pointer` up, so it's located inside of __day

Done

File: src/styles/main.scss (hunk 6 above)
Comment (WaveOfDandelions) [reply src/styles/main.scss]: > Make sure to list all styles that you apply transition to - don't use `all` either.

Ready!

File: src/styles/main.scss (hunk 4 above)
Comment (WaveOfDandelions) [reply src/styles/main.scss]: > This should be inside of day. Inside of loops, put only the properties that are relying on loop - in this case content.

Changed

File: src/styles/main.scss (hunk 3 above)
Comment (WaveOfDandelions) [reply src/styles/main.scss]: > I don't think this two variables make sense

I was apply one padding value

**Review by choeqq — APPROVED** [review]
All changes applied - nice! ✅

**Pull request closed without merging**